scripts/gen-api-gtkdoc.py

```diff
@@ -0,0 +1,411 @@
+			self.annot.append(type.split(":")[1])
+		elif (type == "geany:cb"):
+			self.annot.append("scope notified")
+		elif (type == "geany:cbdata"):
+			self.annot.append("closure")
+		elif (type == "geany:cbfree"):
+			self.annot.append("destroy")
+		elif (type == "geany:transfer") or (type == "geany:element-type") or (type == "geany:scope"):
+			type = type.split(":")[1]
+			self.annot.append("%s %s" % (type, str))
+		elif (type ==  "see"):
+			return "See " + str
+		elif (type ==  "a"):
+			if (str != "NULL"): # FIXME: some of geany does @a NULL
+				return "@" + str
+			else:
+				return str
+		else:
+			return str
+
+		return ""
+
+class At(object):
+	def __init__(self, cb):
+		self.cb = cb
+
+class DoxygenProcess(object):
+	def __init__(self):
+		self.at = None
+
+	# http://stackoverflow.com/questions/4624062/get-all-text-inside-a-tag-in-lxml
+	@staticmethod
+	def stringify_children(node):
+		from lxml.etree import tostring
+		from itertools import chain
+		parts = ([node.text] +
+				list(chain(*([c.text, tostring(c).decode("utf-8"), c.tail] for c in node.getchildren()))) +
+				[node.tail])
+		# filter removes possible Nones in texts and tails
+		return "".join(filter(None, parts))
+
+	def get_program_listing(self, xml):
+		#~ return "--- CODE ---"
+		from lxml.etree import tostring
+		arr = ["", "|[<!-- language=\"C\" -->"]
+		for l in xml.getchildren():
+			if (l.tag == "codeline"):
+				# a codeline is of the form
+				# <highlight class="normal">GeanyDocument<sp/>*doc<sp/>=<sp/>...;</highlight>
+				# <sp/> tags must be replaced with spaces, then just use the text
+				#~ html = self.stringify_children(l)
+				#~ print(etree.HTML(html))
+				h = l.find("highlight")
+				if h is not None:
+					html = tostring(h).decode("utf-8")
+					html = html.replace("<sp/>", " ")
+					arr.append("  " + tostring(etree.HTML(html), method="text").decode("utf-8"))
+		arr.append("]|")
+		return "\n".join(arr)
+
+	def join_annot(self):
+		s = " ".join(map(lambda x: "(%s)" % x, self.at.annot))
+		return s + ": " if s else ""
+
+	def process_element(self, xml):
+		self.at = AtDoc()
+		s = self.__process_element(xml)
+		return s
+
+	def get_extra(self):
+		return self.join_annot()
+
+	def get_return(self):
+		return self.at.retval
+
+	def get_since(self):
+		return self.at.since
+
+	def __process_element(self, xml):
+		s = ""
+
+		if xml.text:
+			s += xml.text
+		for n in xml.getchildren():
+			if n.tag == "emphasis":
+				s += self.at.cb("a", self.__process_element(n))
+			if n.tag == "computeroutput":
+				s += self.at.cb("c", self.__process_element(n))
+			if n.tag == "itemizedlist":
+				s += "\n" + self.__process_element(n)
+			if n.tag == "listitem":
+				s += " - " + self.__process_element(n)
+			if n.tag == "para":
+				s += self.__process_element(n) + "\n"
+			if n.tag == "ref":
+				s += n.text if n.text else ""
+			if n.tag == "simplesect":
+				ss = self.at.cb(n.get("kind"), self.__process_element(n))
+				s += ss if ss + "\n" else ""
```

Comment:
looks wrong, as it's a convoluted way to write `ss` (as adding the `"\n"` in the test makes it evaluate to true).  Shouldn't the `\n` be added on the other value rather than the test?